packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,617 @@
+| Up/Down Arrow keys                           | To move the focus upward/downward in the list of options                                                                            |
+| Left/Right Arrow keys                        | To move the cursor in the editor leftward/rightward and filters the list for the characters from `@` to the current cursor position |
+| Left/Right Arrow keys (cursor away from `@`) | Close the dropdown list popup                                                                                                       |
+| Escape                                       | To close the dropdown if it is opened                                                                                               |
+| Any character                                | To show the filtered list of users for that specific character                                                                      |
+| Space(One time between names)                | To show the filtered list of users that contains a single whitespace in their name                                                  |
+| Space(Two or more trailing white spaces)     | Close the dropdown list popup                                                                                                       |
+| Group of characters                          | To show the filtered list of users for the group characters                                                                         |
+| Group of characters(Not in any option)       | Close the dropdown list popup                                                                                                       |
+|                                              | ----Below are the keyboard interactions when user is selected from the list----                                                     |
+| Backspace                                    | To remove the entire selected name and cursor in the `@` position                                                                   |
+| Shift + Arrow keys                           | To select the mention node                                                                                                          |
+
+_Note_: The `@mention` node is immutable. For an instance, if the user is selected and mention node(`nimble-rich-text-user-mention-view`) is added to the DOM,
+the only option is to either remove the entire name or can add new name after removing the entire name. It is not possible to edit the name, like removing just
+the last name and keeping the first name.
+
+_Mouse interactions for `@mention`_
+
+1. Hovering over each option in the `@mention` dropdown lists will show an indication of the option is selectable.
+2. Left clicking the option will select the hovered option from the list of users.
+3. Clicking outside the editor/Focusing out from the editor will close the popup if it is already open.
+
+_ARIA Roles/Properties for `@mention` components_
```

Comment:
We should figure out the ARIA roles/properties for other display components like the `nimble-rich-text-user-mention-view`. This will be important when a screen reader is reading the contents of a rich text editor / viewer. We want to make sure it speaks the display name, not the ID, for example.

You could view the accessibility tree in dev tools for GitHub or AzDO @mentions to get an idea of what we might want to do here.

Reply:
I have added an ARIA property for the view component. Since they do not have any specific role like `button` or `link` for user mentions, they do not have any specific roles to mention in the node but it contains an `aria-label` with the value of a user name when the screen reader reads the content.

Reply:
Thanks for looking into it. For the listbox and list option roles and attributes, do any of those require unique code for this feature, or is that leveraging existing code on those elements?

Reply:
All the unique codes for each option were leveraged from existing code on those elements.

Reply:
Regarding the role and attributes for the user-mention-view, I'm not sure that setting `aria-label` is the right choice. From reading the [MDN docs for aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) a couple things stand out:
1. "If the label text is available in the DOM, and referencing the DOM content and acceptable user experience, prefer to use aria-labelledby. "
2. "Note: aria-label is intended for use on interactive elements, or elements made to be interactive via other ARIA declarations, when there is no appropriate text visible in the DOM that could be referenced as a label"
2. "Not all elements can be given an accessible name. Neither aria-label nor aria-labelledby should be used with non-interactive elements or inline structural role such as with code, term, or emphasis "

I am doing some more reading now to try to come up with a concrete recommendation but to me this implies:
1. we might not want a label at all (depends on what role we choose)
2. if we do want a label, we should probably use labelledby to point at the text that already exists in the DOM rather than label to provide the same text again.

Reply:
@jattasNI We once again checked with the PO today regarding option 1 which is to show a tooltip when mention node is clicked and they mentioned that it is out of scope for the basic `@mention` support. Since the tooltip is kind of bringing up a whole new component of having all user-related information that involves the complete process from UI design, we thought we had it out of scope for the current feature proposal in this HLD.

> 2. "Note: aria-label is intended for use on interactive elements, or elements made to be interactive via other ARIA declarations when there is no appropriate text visible in the DOM that could be referenced as a label"

Considering the above-mentioned point, I think it is not necessary to have an `aria-label` as well and also not have the label via an `aria-labelledby` because this element is not going to be an interactive element and the same text already displayed in the DOM. 

We think that option 2 can be included but since the showing of tooltip is definitely a potential future scope, we can enable the actual roles like button or link like in other products once we add that feature.

So we are thinking of going with option 3 to treat it as a text for now. We think that even for unsighted users the content will be read by the screen readers which will make the user identify that it is a name mentioned in the editor or viewer. However, this is the high-level thought we had about the accessibility of these nodes. If adding a role like "mark" makes more sense then we could incorporate it.

I have mentioned option 1 as a future scope in the appropriate sections of the HLD to make the necessary updates to accessibility when the feature is added in the future. Please let us know if this decision seems better to proceed forward for now.